### ISSUE TITLE
Lock assets in git submodules

### ISSUE THREAD

**Issue by ArosPrince** [SRombauts/UE4GitPlugin#127]
Hi, this may be an unusual use case, but here's the thing...

In our project we mount some content submodules (I mean git submodules) within the content folder. The point is to reuse our common material library among several projects. However, this GIT plugin obviously does not count with this scenario. So for example, if I lock an asset, it get's locked within the parent repo (the project repo). Well, one problem is that this file does not actually exist in this repo (I am quite surprised that it even get's through). But the biggest issue is that it doesn't really lock the file within the submodule repo, so if you edit the same material from within other project, you still get conflicts.

Would it be possible to update the plugin so that it also handles nested submodules correctly or is it too complicated?

How do you guys reuse materials? Does everyone copy everything over?

BTW I know that Perforce has a nice solution for this, but that's terribly expensive...

**Comment by ccarvalheiro** [comment]
Hi,

We have taken a similar approach for sharing assets between multiple projects, but instead of putting them in the main content, we organize them in plugins so that we choose what we need or don't for each project. That said, we kinda hammered support (according to our workflow) for submodules to our fork of the plugin and some other improvements that I would like to contribute back. A full-blown clean implementation needs to take into account multiple considerations and might not be trivial. We've been really busy and I haven't been able to dedicate time to this and therefore can't promise that will be able to prepare a pull request in the short term.
An alternative might be teaching your artists how to use a git client for committing submodule content.

**Comment by ArosPrince** [comment]
Right, thanks for sharing your experiences. I will give it a second thought...

Committing submodule content is not that difficult even for artists, but locking assets from the command line is :-)

**Comment by budgidiere** [comment]
@ArosPrince Do you get errors when locking submodule files?
